### DISCUSSION TITLE
Snap Camera not appearing as a source to other apps on MacOS Sonoma

### DISCUSSION THREAD

**Discussion by ericlitman** [ptrumpis/snap-camera-server#76] [Q&A] [answered]
Running Sonoma 14.1 on an M2 with the patched Snap Camera app, Snap Camera no longer appears to other apps as a video input source since the upgrade. I reinstalled/patched Snap Camera to no avail.

Any thoughts?

**Comment by ericlitman** [comment]
Still can't seem to figure out how to get the virtual driver to load into other apps. Filters work fine but they'll only display in the in-app preview.

Is anyone on a Mac successfully using this right now?

**Reply by ptrumpis** [reply]
Did you check your privacy settings?

This might also help: https://support.apple.com/HT211130

**Reply by ericlitman** [reply]
Yeah thanks for that. I can view any camera in Snap so it has access to the camera (and mic), but I can't get any other app to see Snap's virtual driver.

**Reply by vitorviesi** [reply]
Same situation here, any solution?

**Reply by duncanjriley** [reply]
Same issue for me. Privacy settings checked and good. Snap Camera works fine on its own. But no other app will recognise it (Zoom/Browser/Google Meet etc)

**Comment by ptrumpis** [comment]
I believe the virtual camera driver will no longer work as of macOS 14 Sonoma.
You might have to downgrade to macOS 13.

Or you need to use the virtual camera driver of [OBS Studio 30.0](https://github.com/obsproject/obs-studio/releases)
- Use Snap Camera application window as input
- Select and crop the application preview area
- And turn that into a virtual webcam

Edit:
See my reply below for possible fix

---

I don't know for sure but Snap Camera could be using a so called DAL plug-in for its virtual driver.
Those will no longer work https://developer.apple.com/videos/play/wwdc2022/10022/

https://github.com/ptrumpis/snap-camera-server/assets/116500225/c9433424-2cb7-4f44-8e19-147a864cea54

**Reply by ptrumpis** [reply, marked as answer]
### Suggested Fix by Apple
[If you can't use your camera or video output device after updating to macOS Sonoma 14.1](https://support.apple.com/en-us/HT213969)

From the support article above:

1. Restart your Mac into Recovery Mode.
   - [Use macOS Recovery on a Mac with Apple silicon](https://support.apple.com/guide/mac-help/mchl82829c17/mac)
   - [Use macOS Recovery on an Intel-based Mac](https://support.apple.com/guide/mac-help/use-macos-recovery-on-an-intel-based-mac-mchl338cf9a8/mac)
3. Click the Options button, then click Continue.
4. From the Utilities menu, choose Terminal.
5. Enter this command:
   ```sh
   system-override legacy-camera-plugins-without-sw-camera-indication=on
   ```
   If you receive the error message: `command not found` you will have to [Turn off FileVault on Mac](https://support.apple.com/guide/mac-help/mchlp2560/14.0/mac/14.0) first.
8. Press Return. Depending on your system configuration, you might be prompted to complete additional steps.
9. Quit Terminal.
10. From the Apple menu, choose Restart.

And check your **Privacy & Security** settings for Camera

**Reply by duncanjriley** [reply]
This worked perfectly thank you! SnapCam now appearing within all software. Note for others - I also had to switch off FileVault encryption before I entered the Terminal command (it failed first time with encryption on)

**Reply by erwiower2** [reply]
Thanks. But now I see in Google Chrome and Microsoft Teams a black screen if snap camera selected. How can I fix that?

**Reply by gernar** [reply]
Did you found the solution? I have the same problem, but my snap camera gives back screen in obs, and doesn’t even appear in google meets etc 

P.S.: fixed it in obs by changing output resolution of the source

**Comment by Oph104** [comment]
Thanks , this helped me. There will e another update for Mac Sonoma 14.1.2.  Has anyone had any issues since the update? I have not updated yet.

**Comment by ferso** [comment]
It doesn’t work anymore on Sonoma 14.3

**Reply by ptrumpis** [reply]
Did you try to apply the suggested fix again after updating to 14.3?

Anyway here some more help articles with possible solutions:
[Mac Camera Not Working after Sonoma Update? 11 Quick Fixes](https://www.drbuho.com/how-to/fix-mac-camera-not-working)
[How to Fix Camera Not Working in macOS Sonoma](https://www.macobserver.com/tips/how-to/webcam-not-working-macos/)

**Reply by Oph104** [reply]
OBS studio? I looked on the solutions and did not see anything on there, maybe I overlooked it?

**Reply by coderfin** [reply]
https://github.com/ptrumpis/snap-camera-server/discussions/76#discussioncomment-7470839

**Reply by Oph104** [reply]
Thank you! :)

**Reply by Ivich25** [reply]
OBS works. I was a bit stuck at first but it works a treat, even on the new update 14.4.

**Comment by neocody** [comment]
The system-override command is not found. I'm on 14.1 and I have tried every available help step to fix this. I disabled filevault. I have used disk utility to make sure the APFS drive is mounted, and it was. I have no idea why I keep getting command not found.
![IMG_8135](https://github.com/ptrumpis/snap-camera-server/assets/25426121/1a0baab6-bad9-4c08-8ab5-60e1303fae9b)

**Reply by ptrumpis** [reply]
You need to [Turn off FileVault on Mac](https://support.apple.com/guide/mac-help/mchlp2560/14.0/mac/14.0).
The command should then work.

**Comment by neocody** [comment]
It is off.
